woocommerce/woocommerce-ios
# External products: external link & button text

"editable product link" for #2200 

## Changes

- Encoded `externalURL` and `buttonText` for updating an external product
- Added a function to update `externalURL` and `buttonText` in `ProductFormViewModel` + test case
- Created `ProductExternalLinkViewController` that enabled editing the external URL and button text fields for an external product
- Created an enum `TextFieldContentAlignment` for different ways of text field content alignment for the text and placeholder (I was debating between this name and `TextFieldTextAlignment`, lemme know if you think another name would sound more clear!)
  - Because `NSTextAlignment` only has `left`/`right` instead of `leading`/`trailing` that takes LTR/RTL layout into account, this enum is mapped to `NSTextAlignment` based on the app layout
- Added `keyboardType` and `textFieldAlignment` parameters to `TitleAndTextFieldTableViewCell` for customizations in the external link screen
- Enabled editing external link in `ProductFormViewController`

## Testing

- Go to the Products tab
- Tap on an external product
- Tap on the Product link row --> should see the screen with two text fields for external URL and button text as in the screenshots
- Edit the external URL and button text
- (If not on iOS 13.4 simulator/device) Tap "<" in the navigation bar --> a discard changes prompt should be shown
- Tap "Cancel" to keep editing
- Tap "Done" to save changes
- Tap "Update" --> the external URL and button text should be updated remotely

## Example screenshots

![simulator](https://user-images.githubusercontent.com/1945542/84986133-973c8200-b170-11ea-8162-27d3d0751681.gif)


empty state | filled state
-- | --
![Simulator Screen Shot - iPhone 11 - 2020-06-18 at 10 34 25](https://user-images.githubusercontent.com/1945542/84972500-8e888380-b151-11ea-84b0-b2e687b50d1c.png) | ![Simulator Screen Shot - iPhone 11 - 2020-06-18 at 10 34 48](https://user-images.githubusercontent.com/1945542/84972506-91837400-b151-11ea-9700-ae38a2317e04.png)
![Simulator Screen Shot - iPhone 11 - 2020-06-18 at 10 35 33](https://user-images.githubusercontent.com/1945542/84972509-934d3780-b151-11ea-820f-e92116cdec95.png) | ![Simulator Screen Shot - iPhone 11 - 2020-06-18 at 10 35 28](https://user-images.githubusercontent.com/1945542/84972508-92b4a100-b151-11ea-8428-97a1c37b9b4d.png)


![Simulator Screen Shot - iPhone 11 - 2020-06-18 at 10 33 48](https://user-images.githubusercontent.com/1945542/84972519-97795500-b151-11ea-98a1-c95dc28a968d.png)


Update release notes:
